frontend/test/metabase/scenarios/dashboard-filters/dashboard-filters-source.cy.spec.js

```diff
@@ -92,6 +94,40 @@ describe("scenarios > dashboard > filters", () => {
     saveDashboard();
     filterDashboard();
   });
+
+  it("should result in a warning being shown when archiving a question it uses", () => {
+    cy.intercept("POST", "/api/dashboard/**/query").as("getCardQuery");
+
+    cy.createQuestion(structuredQuestionDetails, {
+      wrapId: true,
+      idAlias: "structuredQuestionId",
+    });
+    cy.createQuestionAndDashboard({
+      questionDetails: dashboardQuestionDetails,
+    }).then(({ body: { dashboard_id } }) => {
+      visitDashboard(dashboard_id);
+    });
+
+    editDashboard();
+    setFilter("Text or Category", "Dropdown");
+    mapFilterToQuestion();
+    editDropdown();
+    setupStructuredQuestionSource();
+    saveDashboard();
+
+    cy.intercept("GET", "/api/collection/root/items**").as("getItems");
+
+    cy.get("@structuredQuestionId").then(question_id => {
+      visitQuestion(question_id);
+      openQuestionActions();
+      cy.findByTestId("archive-button").click();
+      cy.get(".Modal")
```

Comment:
Why not `modal().within(() => `?

Reply:
No reason, I cargo-culted from an earlier test. I'll change it